filebeat/tests/system/test_processors.py

```diff
@@ -300,6 +300,254 @@ def test_decode_csv_fields_all_options(self):
+
+    def test_javascript_processor_add_process_metadata(self):
+        """
+        Check JS processor with add_process_metadata
+        """
+
+        self._test_javascript_processor_with_source("""\'var processor = require("processor");
+var addProcessMetadata = new processor.AddProcessMetadata({
+    match_pids: "process.pid",
+    ignore_missing: false,
+    overwrite_keys: true,
+});
+function process(evt) {
+    addProcessMetadata.Run(evt);
+}\'
+""")
+
+        output = self.read_output()
+        for evt in output:
+            assert "error.message" in evt
+            assert evt["error.message"] == "GoError: none of the fields in match_pids found in the event"
+
+    def test_javascript_processor_community_id(self):
+        """
+        Check JS processor with community_id
+        """
+
+        self._test_javascript_processor_with_source("""\'var processor = require("processor");
+var communityID = new processor.CommunityID();
+function process(evt) {
+    communityID.Run(evt);
+}\'
+""", True)
+
+        output = self.read_output()
+        for evt in output:
+            assert "network.community_id" in evt
+            assert evt["network.community_id"] == "1:15+Ly6HsDg0sJdTmNktf6rko+os="
+
+    def test_javascript_processor_copy_fields(self):
+        """
+        Check JS processor with copy_fields
+        """
+
+        self._test_javascript_processor_with_source("""\'var processor = require("processor");
+var copy = new processor.CopyFields({
+    fields: [
+    {from: "message", to: "log.original"},
+    ],
+});
+function process(evt) {
+    copy.Run(evt);
+}\'
+""")
+
+        output = self.read_output()
+        for evt in output:
+            assert "log.original" in evt
+            assert evt["log.original"] == evt["message"]
+
+    def test_javascript_processor_decode_json_fields(self):
+        """
+        Check JS processor with decode_json_fields
+        """
+
+        self._test_javascript_processor_with_source("""\'var processor = require("processor");
+var decodeJSON = new processor.DecodeJSONFields({
+    fields: ["json_message_to_decode"],
+    target: "json_message",
+});
+function process(evt) {
+    decodeJSON.Run(evt);
+}\'
+""", True)
+
+        output = self.read_output()
+        for evt in output:
+            assert "json_message.key" in evt
+            assert evt["json_message.key"] == "hello"
+
+    def test_javascript_processor_dissect(self):
+        """
+        Check JS processor with dissect
+        """
+
+        self._test_javascript_processor_with_source("""\'var processor = require("processor");
+var chopLog = new processor.Dissect({
+    tokenizer: "key=%{key}",
+    field: "dissect_message",
+});
+function process(evt) {
+    chopLog.Run(evt);
+}\'
+""", True)
+
+        output = self.read_output()
+        for evt in output:
+            assert "dissect.key" in evt
+            assert evt["dissect.key"] == "hello"
+
+    def test_javascript_processor_dns(self):
```

Comment:
The test for the `reverse` processor was skipped on Windows. As it is not failing I wonder if we should add some assertion to this test.